scripts/check

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+#
+# This script verifies that the main binary can be build, tests pass and files
+# are properly formatted. It is used to verify code changes in pull requests.
+#
+# Requirements:
+#   - Go 1.5 or later. If using 1.5, set GO15VENDOREXPERIMENT=1. If using 1.6 or
+#     later, that is not necessary.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Make sure that all Go files are properly formatted.
+GOFMT_DIFF="$(find . -path ./vendor -prune -o -name '*.go' -exec gofmt -s -d {} \;)"
+if [ -n "${GOFMT_DIFF}" ]; then
+  echo >&2 $'----------------------\nSome files need gofmt:\n----------------------'
+  echo >&2 "${GOFMT_DIFF}"
+  exit 1
+fi
+
+# Make sure the main binary is buildable.
+go build -o /dev/null
```

Comment:
tip of the day: never write bash in a hurry... or, better yet, never write a shell script!